modules/sql/src/main/resources/db/migration/V012__Static_Config.sql

```diff
@@ -0,0 +1,54 @@
+--
+-- This migration adds support for separating the static vs. dynamic (step)
+-- parts of the instrument configuration.  Each observation must have an
+-- instrument and its static configuration.
+--
+
+-- At this point we have just one instrument, F2, and it has just one bit of
+-- static configuration: the mos_preimaging flag.  In this migration we'll move
+-- that flag from it's previous home in the dynamic (step) configuration to a
+-- new static_f2 table.
+ALTER TABLE step_f2
+  DROP COLUMN mos_preimaging;
+
+-- static_config houses a unique id for each static configuation.  Though the
+-- id is sufficient for a primary key, we add the corresponding instrument as a
+-- discriminant.  Over in observation, the static_id and instrument will be a
+-- non-null foreign key into this table.  That ensures that the observation has
+-- a static component and that it uses the same instrument as the observation
+-- itself.
+CREATE TABLE static_config (
+    static_id  SERIAL,
+    instrument identifier NOT NULL REFERENCES e_instrument,
+    PRIMARY KEY (static_id, instrument)
+);
+
+ALTER TABLE static_config OWNER TO postgres;
+
+-- Each instrument will have its own static configuration table.  This is the
+-- static configuration table for F2, which has a payload of a single boolean
+-- mos-preimaging flag.  We have the same (static_id, instrument) foreign key
+-- into the static_config table and furthermore guarantee that the instrument
+-- is always Flamingos2.
+CREATE TABLE static_f2 (
+    static_id       integer,
+    instrument      identifier NOT NULL,
+    mos_preimaging  boolean    NOT NULL,
+    PRIMARY KEY (static_id, instrument),
+    FOREIGN KEY (static_id, instrument) REFERENCES static_config ON DELETE CASCADE,
+    CONSTRAINT is_f2 CHECK (instrument = 'Flamingos2')
```

Comment:
This is what we discussed face to face. It seems fine to me to have this check there